### ISSUE TITLE
Fork doesn't notice changed files

### ISSUE THREAD

**Issue by kingtypo** [ForkIssues/TrackerWin#619]
The file changes take time to appear in the 'Changes' context.  A manual refresh button would really help in refreshing the file changes. It would be even better if an automatic refresh trigger is set every time the context is focused on Changes & and an interval-based refresh when already on the context.

**Comment by decadence** [comment]
F5 triggers refresh

**Comment by felixhirt** [comment]
pretty sure the refresh also happens when the fork window receives focus

**Comment by DanPristupov** [comment]
I found a bug that once a day Fork doesn't notice the focus and skips the refresh. This will be fixed in the next update (1.44).

Sometimes bugs can be really strange 🙂

**Comment by DanPristupov** [comment]
Have anybody seen the problem since 1.44?

**Comment by jerone** [comment]
> Have anybody seen the problem since 1.44?

Yes :( Repeatedly I'm experiencing this issue. Waiting a few seconds did not always work. Pressing F5 always works.

**Comment by vitalyster** [comment]
Experiencing the same on 1.49.1.0

**Comment by decadence** [comment]
Unfortunately I have this problem on the latest 1.51.2.0 sometimes and already got used to F5.

**Comment by DanPristupov** [comment]
I fixed some related issues in 1.55. Could you check if the problem persists there?

**Comment by decadence** [comment]
At first glance it's fixed but need more time to confirm that

**Comment by octocat-mona** [comment]
Doesn't seem to work for me (v 1.55.1.0).
Pressing F5 does refresh immediately but just switching to Fork doesn't refresh.

**Comment by DanPristupov** [comment]
> Pressing F5 does refresh immediately but just switching to Fork doesn't refresh.

Do you mean it *never* refreshes? Or it doesn't refresh sometimes?

**Comment by octocat-mona** [comment]
I have never seen it refresh on its own when I switch to Fork.
What I do see is that:

- When Fork is in the foreground but does not have active focus it does not refresh until it gets focus
- When Fork is in the background and I switch to it it does not refresh, but it does refresh when I minimize and bring back up Fork again
- Of course it refreshes when pressing F5 manually

Does this help?

**Comment by vitalyster** [comment]
Still actual for 1.55.5.0

**Comment by DanPristupov** [comment]
@vitalyster could you check the latest 'Application Window Activated: skip' message in `%localappdata%\fork\logs\fork.log`? It must contain a reason for skipping the update.

**Comment by decadence** [comment]
No. I'm on 1.55.5.0 and now I had to press F5 to see the changes. I think it's fixed partially because it happens less often.

**Comment by DanPristupov** [comment]
@decadence could you check the log when this happens next time please? You can send whole log file to my email: support@fork.dev

**Comment by decadence** [comment]
@DanPristupov done

**Comment by egorpavlikhin** [comment]
@DanPristupov Can you add a button or menu item to force refresh? Currently have to use Rider to commit changes, because it does have such button, so I can just click it before committing.

**Comment by felixhirt** [comment]
@egorpavlikhin F5 should trigger a manual refresh. Does that not work as well?
